test/gtest/kernel_tuning_net.cpp

```diff
@@ -86,131 +98,157 @@ std::vector<KernelTuningNetTestCase> GetConvHipIgemmGroupWrwXdlopsTestCases()
-        auto test_case                             = GetParam();
-        miopen::TensorDescriptor input_tensor_desc = miopen::TensorDescriptor(
+        auto test_case = GetParam();
+
+        auto&& handle = get_handle();
+        miopen::ExecutionContext ctx(&handle);
+
+        if(test_case.arch != ctx.GetStream().GetDeviceName())
+            GTEST_SKIP();
+
+        auto input_tensor_desc = miopen::TensorDescriptor(
             test_case.data_type, test_case.layout, test_case.conv.GetInput());
-        miopen::TensorDescriptor weights_tensor_desc = miopen::TensorDescriptor(
+
+        auto weights_tensor_desc = miopen::TensorDescriptor(
             test_case.data_type, test_case.layout, test_case.conv.GetWeights());
-        auto conv_desc                       = test_case.conv.GetConv();
-        miopen::TensorDescriptor output_desc = conv_desc.GetForwardOutputTensor(
+
+        auto conv_desc = test_case.conv.GetConv();
+
+        auto output_desc = conv_desc.GetForwardOutputTensor(
             input_tensor_desc, weights_tensor_desc, test_case.data_type);
-        problem  = (test_case.direction == miopen::conv::Direction::Forward)
-                       ? miopen::conv::ProblemDescription(input_tensor_desc,
-                                                         weights_tensor_desc,
-                                                         output_desc,
-                                                         conv_desc,
-                                                         test_case.direction)
-                       : miopen::conv::ProblemDescription(output_desc,
-                                                         weights_tensor_desc,
-                                                         input_tensor_desc,
-                                                         conv_desc,
-                                                         test_case.direction);
-        expected = test_case.expected_config;
-        arch     = test_case.arch;
+
+        auto problem = (test_case.direction == miopen::conv::Direction::Forward)
+                           ? miopen::conv::ProblemDescription(input_tensor_desc,
+                                                              weights_tensor_desc,
+                                                              output_desc,
+                                                              conv_desc,
+                                                              test_case.direction)
+                           : miopen::conv::ProblemDescription(output_desc,
+                                                              weights_tensor_desc,
+                                                              input_tensor_desc,
+                                                              conv_desc,
+                                                              test_case.direction);
+
+        Solver perf_config;
+        ASSERT_TRUE(perf_config.IsModelApplicable(ctx, problem));
```

Comment:
Looks like we used to skip if the model wasn't applicable, and now we just assert for it.
Is the intent that we only create these when the models are applicable, and it should fail if it doesn't match?

Reply:
It does not make sense to skip it. I expected this question)
Let's imagine that it's not applicable, skipped and... we don't have any tests to check AI tuning anymore. Silently.

Reply:
Makes sense to me, but wanted to confirm!